middleware.ts

```diff
@@ -8,7 +8,14 @@ export async function middleware(req: NextRequest) {
   const supabase = createMiddlewareClient({ req, res });
 
   // Refresh session if expired - required for Server Components
-  await supabase.auth.getUser();
+  const {
+    data: { user },
+  } = await supabase.auth.getUser();
+
+  // Redirect to login page if no user
+  if (!user && !req.url.includes('login')) {
```

Comment:
Can we change this to `.nextUrl.pathname.startsWith("/login")`? Should be a little safer than includes.

Reply:
good shout! have updated